src/common/util.js

```diff
@@ -161,7 +210,16 @@ function hasNewlineInRange(text, start, end) {
 }
 
 // Note: this function doesn't ignore leading comments unlike isNextLineEmpty
+/**
+ * @template N
+ * @param {string} text
+ * @param {N} node
+ * @param {(node: N) => number} locStart
+ */
 function isPreviousLineEmpty(text, node, locStart) {
+  /**
+   * @type {number | false}
+   */
   let idx = locStart(node) - 1;
```

Comment:
Please don't add types to local variables. They are inferred by the typechecker and are not part of API.

Reply:
This is needed to tell TypeScript that `idx` won’t just be a number, but can also be `false`?

Reply:
Right, sorry.